auth/office365.go

```diff
@@ -45,10 +46,12 @@ func Office365(addUser func(user, token string), clientID, clientSecret, domain
 		}
 
 		if isInDomain(user, domain) {
-			token := strId()
-			addUser(user, token)
-
-			http.Redirect(w, r, "/?user="+user+"&token="+token, http.StatusFound)
+			idToken, err := addUser(user)
+			if err != nil {
+				http.Redirect(w, r, "/?error=could_not_create_user", http.StatusFound)
+			} else {
+				http.Redirect(w, r, "/?token="+idToken, http.StatusFound)
+			}
```

Comment:
should probably extract all of this redirect logic somewhere else, since it will be the same for any auth provider